src/main/java/seedu/address/model/person/TagContainsKeywordsPredicate.java

```diff
@@ -0,0 +1,54 @@
+package seedu.address.model.person;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.function.Predicate;
+
+import seedu.address.commons.util.StringUtil;
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.model.tag.Tag;
+
+/**
+ * Tests that a {@code Person}'s {@code Tags} matches any of the keywords given.
+ */
+public class TagContainsKeywordsPredicate implements Predicate<Person> {
+    private final List<String> keywords;
+
+    public TagContainsKeywordsPredicate(List<String> keywords) {
+        this.keywords = keywords;
+    }
+
+    @Override
+    public boolean test(Person person) {
+        Set<Tag> tags = person.getTags();
+        return tags.stream().anyMatch(tag ->
+            keywords.stream().anyMatch(keyword ->
+                StringUtil.containsSubstringIgnoreCase(tag.tagName, keyword))
+        );
```

Comment:
i think this might be O(n^2) but it probably shouldn't matter because there probably shouldn't be too many tags or keywords?

Reply:
Yeah but I think maybe this is only a big concern if the data is large in size, but in a single find command I think it is safe to assume that it won't degrade the performance too much? Actually I am not sure if the second nested lambda is necessary but will try to manually test with some bigger input?

Reply:
yup agree, probably can ignore

if you want to do O(n) then probably some set intersection between `tags` and `keywords`